### PR TITLE
Convert closures to arrow functions

### PR DESCRIPTION
**Pull request by duyphuongn**
- [x] Changes have been thoroughly tested to not break existing functionality.

---

This PR converts traditional closures to arrow functions, resulting in cleaner and consistent code style throughout Filament code base.

Used tool: 

- [Rector 0.17.3](https://github.com/rectorphp/rector) (with [`ClosureToArrowFunctionRector`](https://github.com/rectorphp/rector/blob/main/rules/Php74/Rector/Closure/ClosureToArrowFunctionRector.php) rule) for refactoring
- Laravel Pint 1.10.6 for formatting

Each change was verified manually before being applied.

### CODE REVIEW

**Comment by wychoong** [comment]
i think the readability is not as good

**Comment by duyphuongn** [comment]
> i think the readability is not as good

Most closures in code base are already written using arrow functions, even in [documentation](https://filamentphp.com/docs/3.x/panels/resources/listing-records). This PR converts remaining traditional closures to make coding style consistent.

For readability, I don't think it's a big issue because we spend most time working without custom code, not the packages.

File: packages/forms/src/Components/Concerns/CanBeValidated.php [hunk 1]
```diff
@@ -262,9 +262,7 @@ public function macAddress(bool | Closure $condition = true): static
 
     public function multipleOf(int | Closure $value): static
     {
-        $this->rule(static function (Field $component) use ($value) {
-            return 'multiple_of:' . $component->evaluate($value);
-        }, static fn (Field $component): bool => filled($component->evaluate($value)));
+        $this->rule(static fn (Field $component) => 'multiple_of:' . $component->evaluate($value), static fn (Field $component): bool => filled($component->evaluate($value)));
```
Comment (wychoong) [inline packages/forms/src/Components/Concerns/CanBeValidated.php:265]: example of bad readability

File: packages/forms/src/Components/Concerns/CanBeAutocompleted.php [hunk 2]
```diff
@@ -21,9 +21,7 @@ public function autocomplete(bool | string | Closure | null $autocomplete = true
      */
     public function disableAutocomplete(bool | Closure $condition = true): static
     {
-        $this->autocomplete(static function (Field $component) use ($condition): ?bool {
-            return $component->evaluate($condition) ? false : null;
-        });
+        $this->autocomplete(static fn (Field $component): ?bool => $component->evaluate($condition) ? false : null);
```
Comment (wychoong) [inline packages/forms/src/Components/Concerns/CanBeAutocompleted.php:24]: simple logic that is easy to read, and no harm to convert into arrow function

**Comment by wychoong** [comment]
> This PR converts remaining traditional closures to make coding style consistent.

using arrow function is just convenience

> For readability, I don't think it's a big issue because we spend most time working without custom code, not the packages.

you are not wrong, but there are times code dive helps to understand how it works in the package to build own app/suggest features/bug fix etc.

**Comment by danharrin** [comment]
> For readability, I don't think it's a big issue because we spend most time working without custom code, not the packages.

Speak for yourself, I have to read it and maintain it 😅 I wrote full closures in these places for readability, so the lines aren't too long

Sorry, but not gonna merge this one

**Pull request closed without merging**